File: JavaScript/routes/timerRequests.js
var express = require('express');
var router = express.Router();
const AllTimers=require("../models/taskTimer.js")

const allTimers =new AllTimers()
allTimers.createTimer("first")
allTimers.createTimer("second")
allTimers.startTimer("first")
allTimers.pauseTimer("first")
console.log(allTimers.showAllTimers())

/* GET users listing. */
router.get('/', function(req, res, next) {
    try{
        //CAN MOVE COMMAND INTERPRETER TO FRONTEND LATER, SEND IN REQUEST CLEAN
        const userSentence = req.body
        console.log(userSentence)
        const userCommand = userSentence.command.split(" ")[0]
        const userCommandTarget = userSentence.command.split(" ")[1]
        console.log(userCommand, userCommandTarget)

        // TILL HERE

        switch (userCommand) {
            case "start":
                res.json({message:allTimers.startTimer(userCommandTarget)})
                break;
            case "create":
                res.json({message:allTimers.createTimer(userCommandTarget)})
                break;
            case "pause":
                res.json({message:allTimers.pauseTimer(userCommandTarget)})
                break;
            case "reset":
                res.json({message:allTimers.resetTimer(userCommandTarget)})
                break;
            case "stop":
                res.json({message:allTimers.stopTimer(userCommandTarget)})
                break;
            case "getStatus":
                res.json({message: "Status of timer: " + allTimers.getStatusTimer(userCommandTarget)})
                break;
            case "getElapsedTime":
                res.json({message: "Last action happened on" ,toIsoString:allTimers.getElapsedTimeTimer(userCommandTarget)})
                break;
            case "getUid":
                res.json({message: "Unique ID of timer: " + allTimers.getUidTimer(userCommandTarget)})
                break;
            case "showAllTimers":
                res.json({message: "List of all timers:" + allTimers.showAllTimers()})
                break;
            default:
                res.status(422)
                res.send("Bad input")
        }
    }catch (e){
        console.log(e)
        console.log("USAUUDSA")
        res.status(422)
        res.send({message:e.toString()})
    }
    res.send();
});

module.exports = router;

File: Python/routes/timerRequests.py
from flask import Blueprint, request, jsonify
from models.TaskTimer import AllTimers

timer_requests_bp = Blueprint('timer_requests', __name__)

all_timers = AllTimers()

@timer_requests_bp.route('/timers', methods=['GET'])
def handle_timer_command():
    user_sentence = request.json
    if not user_sentence or 'command' not in user_sentence:
        return jsonify({"error": "Bad input"}), 422

    user_command = user_sentence['command'].split(" ")[0]
    user_command_target = user_sentence['command'].split(" ")[1] if len(
        user_sentence['command'].split(" ")) > 1 else None

    try:
        if user_command == "start":
            all_timers.startTimer(user_command_target)
        elif user_command == "create":
            all_timers.createTimer(user_command_target)
        elif user_command == "pause":
            all_timers.pauseTimer(user_command_target)
        elif user_command == "reset":
            all_timers.resetTimer(user_command_target)
        elif user_command == "getStatus":
            return jsonify({"message": "Status of timer: " + all_timers.getStatusTimer(user_command_target)})
        elif user_command == "getElapsedTime":
            return jsonify({"message": "Elapsed time since last action: " + all_timers.getElapsedTimeTimer(user_command_target)})
        elif user_command == "getUid":
            return jsonify({"message": "Unique ID of timer: " + all_timers.getUidTimer(user_command_target)})
        elif user_command == "showAllTimers":
            return jsonify({"message": "List of all timers: " + all_timers.showAllTimers()})
        else:
            return jsonify({"error": "Invalid command"}), 422
    except Exception as e:
        return jsonify({"error": str(e)}), 500
